# Task 1: Member 
# Texas as specific state

In [41]:
import pandas as pd
import numpy as np
import math
import matplotlib.pyplot as plt
import plotly.express as px
import plotly
import plotly.graph_objs as go

In [42]:
import os

if not os.path.exists("images"):
    os.mkdir("images")

In [43]:
#read super Covid-19 dataframe with state as index. 
Read = pd.read_csv("../../../data/output/covid.csv")
df = Read.drop(["countyFIPS","stateFIPS","Unnamed: 0"], axis=1)
df = df.drop(df.loc[df['County Name'] == "Statewide Unallocated"].index)


## Get total amount of Cases and Deaths per state.

In [44]:
#groups the column by States, and adds up cases of each county per state.
States = df.groupby(['State']).sum()

### Find States with similar population and compare them. 

In [45]:
#Find states with similar Population.
#We choose Texas so we find other states with similar population(CA,Fl,NY, and PA)
States.sort_values(by=['population'],ascending = False)

population  1/22/20_x  1/23/20_x  1/24/20_x  1/25/20_x  1/26/20_x  \
State                                                                      
CA       39512223          0          0          0          0          2   
TX       28995881          0          0          0          0          0   
FL       21477737          0          0          0          0          0   
NY       19453561          0          0          0          0          0   
PA       12801989          0          0          0          0          0   
IL       12671821          0          0          1          1          1   
OH       11689100          0          0          0          0          0   
GA       10617423          0          0          0          0          0   
NC       10488084          0          0          0          0          0   
MI        9986857          0          0          0          0          0   
NJ        8882190          0          0          0          0          0   
VA        8535519          0          0          0          0          0   
WA        7614893          1          1          1          1          1   
AZ        7278717          0          0          0          0          1   
MA        6892503          0          0          0          0          0   
TN        6829174          0          0          0          0          0   
IN        6732219          0          0          0          0          0   
MO        6137428          0          0          0          0          0   
MD        6045680          0          0          0          0          0   
WI        5822434          0          0          0          0          0   
CO        5758736          0          0          0          0          0   
MN        5639632          0          0          0          0          0   
SC        5148714          0          0          0          0          0   
AL        4903185          0          0          0          0          0   
LA        4648794          0          0          0          0          0   
KY        4467673          0          0          0          0          0   
OR        4217737          0          0          0          0          0   
OK        3956971          0          0          0          0          0   
CT        3565287          0          0          0          0          0   
UT        3205958          0          0          0          0          0   
IA        3155070          0          0          0          0          0   
NV        3080156          0          0          0          0          0   
AR        3017804          0          0          0          0          0   
MS        2976149          0          0          0          0          0   
KS        2913314          0          0          0          0          0   
NM        2096829          0          0          0          0          0   
NE        1934408          0          0          0          0          0   
WV        1792147          0          0          0          0          0   
ID        1787065          0          0          0          0          0   
HI        1415872          0          0          0          0          0   
NH        1359711          0          0          0          0          0   
ME        1344212          0          0          0          0          0   
MT        1068778          0          0          0          0          0   
RI        1059361          0          0          0          0          0   
DE         973764          0          0          0          0          0   
SD         884659          0          0          0          0          0   
ND         762062          0          0          0          0          0   
AK         731545          0          0          0          0          0   
DC         705749          0          0          0          0          0   
VT         623989          0          0          0          0          0   
WY         578759          0          0          0     

## Choose States 

In [46]:
array = ['TX','CA','FL','NY','PA','NC']
MyStates = States.loc[States.index.isin(array)]
MyStates

population  1/22/20_x  1/23/20_x  1/24/20_x  1/25/20_x  1/26/20_x  \
State                                                                      
CA       39512223          0          0          0          0          2   
FL       21477737          0          0          0          0          0   
NC       10488084          0          0          0          0          0   
NY       19453561          0          0          0          0          0   
PA       12801989          0          0          0          0          0   
TX       28995881          0          0          0          0          0   

       1/27/20_x  1/28/20_x  1/29/20_x  1/30/20_x  ...  10/12/20_y  \
State                                              ...               
CA             3          3          4          4  ...       16611   
FL             0          0          0          0  ...       15408   
NC             0          0          0          0  ...        3773   
NY             0          0          0          0  ...       32982   
PA             0          0          0          0  ...        8368   
TX             0          0          0          0  ...       16550   

       10/13/20_y  10/14/20_y  10/15/20_y  10/16/20_y  10/17/20_y  10/18/20_y  \
State                                                                           
CA          16677       16771       16850       16920       16954       16990   
FL          15529       15594       15734       15827       15917       15966   
NC           3814        3854        3874        3906        3923        3934   
NY          32993       33001       33020       33033       33045       33060   
PA           8368        8410        8431        8457        8466        8466   
TX          16614       16709       16803       16894       16975       17005   

       10/19/20_y  10/20/20_y  10/21/20_y  
State                                      
CA          17012       17049       17214  
FL          16019       16105       16210  
NC           3939        3992        3992  
NY          33077       33087       33092  
PA           8500        8533        8562  
TX          17013       17078       17192  

[6 rows x 549 columns]

In [47]:
#get Population value of States to use later on. 
TX_Pop = MyStates.iloc[5][0]
CA_Pop = MyStates.iloc[0][0]
FL_Pop = MyStates.iloc[1][0]
NC_Pop = MyStates.iloc[2][0]
NY_Pop = MyStates.iloc[3][0]
PA_Pop = MyStates.iloc[4][0]

## Seperate Cases and Deaths of Specific State

In [48]:
State_cases = MyStates.filter(regex = "x").T.rename_axis('Date')
State_deaths = MyStates.filter(regex = "y").T.rename_axis('Date')

In [49]:
#Display State Cases
State_cases.tail()

State           CA      FL      NC      NY      PA      TX
Date                                                      
10/17/20_x  873404  750518  243695  482960  180943  831870
10/18/20_x  874795  753046  245994  484350  182322  837660
10/19/20_x  879572  754742  247168  485348  183315  840216
10/20/20_x  883574  758394  248750  486549  184872  845283
10/21/20_x  886954  760621  248750  488575  186297  850740

In [50]:
#Display State Deaths
State_deaths.tail()

State          CA     FL    NC     NY    PA     TX
Date                                              
10/17/20_y  16954  15917  3923  33045  8466  16975
10/18/20_y  16990  15966  3934  33060  8466  17005
10/19/20_y  17012  16019  3939  33077  8500  17013
10/20/20_y  17049  16105  3992  33087  8533  17078
10/21/20_y  17214  16210  3992  33092  8562  17192

## Calculate New Cases and New Deaths of States

In [51]:
#Calculates differences of cases and deaths of each day (New Cases & New Deaths)
New_State_Cases = State_cases.diff()
New_State_Deaths = State_deaths.diff()

print(New_State_Cases)
print(New_State_Deaths)

New_State_Cases.to_csv("Data/State_cases.csv")
New_State_Deaths.to_csv("Data/State_deaths.csv")

State           CA      FL      NC      NY      PA      TX
Date                                                      
1/22/20_x      NaN     NaN     NaN     NaN     NaN     NaN
1/23/20_x      0.0     0.0     0.0     0.0     0.0     0.0
1/24/20_x      0.0     0.0     0.0     0.0     0.0     0.0
1/25/20_x      0.0     0.0     0.0     0.0     0.0     0.0
1/26/20_x      2.0     0.0     0.0     0.0     0.0     0.0
...            ...     ...     ...     ...     ...     ...
10/17/20_x  1957.0  4029.0  2072.0  1784.0  1857.0  5079.0
10/18/20_x  1391.0  2528.0  2299.0  1390.0  1379.0  5790.0
10/19/20_x  4777.0  1696.0  1174.0   998.0   993.0  2556.0
10/20/20_x  4002.0  3652.0  1582.0  1201.0  1557.0  5067.0
10/21/20_x  3380.0  2227.0     0.0  2026.0  1425.0  5457.0

[274 rows x 6 columns]
State          CA     FL    NC    NY    PA     TX
Date                                             
1/22/20_y     NaN    NaN   NaN   NaN   NaN    NaN
1/23/20_y     0.0    0.0   0.0   0.0   0.0    0.0
1/24/20_y

## Normalize by population (1 million)

In [12]:
#calculate and Normalize dataset to cases per 1,000,000 people. 
norm_val = 1_000_000

#Normalize cases
Norm_MyStateCases = New_State_Cases.assign(CA = lambda x: (x['CA'] / CA_Pop) * norm_val).round(0)
Norm_MyStateCases = Norm_MyStateCases.assign(FL = lambda x: (x['FL'] / FL_Pop) * norm_val).round(0)
Norm_MyStateCases = Norm_MyStateCases.assign(NC = lambda x: (x['NC'] / NC_Pop) * norm_val).round(0)
Norm_MyStateCases = Norm_MyStateCases.assign(NY = lambda x: (x['NY'] / NY_Pop ) * norm_val).round(0)
Norm_MyStateCases = Norm_MyStateCases.assign(PA = lambda x: (x['PA'] / PA_Pop ) * norm_val).round(0)
Norm_MyStateCases = Norm_MyStateCases.assign(TX = lambda x: (x['TX'] / TX_Pop ) * norm_val).round(0)

#Normalize deaths 
Norm_MyStateDeaths = New_State_Deaths.assign(CA = lambda x: (x['CA'] / CA_Pop) * norm_val).round(0)
Norm_MyStateDeaths = Norm_MyStateDeaths.assign(FL = lambda x: (x['FL'] / FL_Pop) * norm_val).round(0)
Norm_MyStateDeaths = Norm_MyStateDeaths.assign(NC = lambda x: (x['NC'] / NC_Pop) * norm_val).round(0)
Norm_MyStateDeaths = Norm_MyStateDeaths.assign(NY = lambda x: (x['NY'] / NY_Pop ) * norm_val).round(0)
Norm_MyStateDeaths = Norm_MyStateDeaths.assign(PA = lambda x: (x['PA'] / PA_Pop ) * norm_val).round(0)
Norm_MyStateDeaths = Norm_MyStateDeaths.assign(TX = lambda x: (x['TX'] / TX_Pop ) * norm_val).round(0)

print(Norm_MyStateCases.tail())
print(Norm_MyStateDeaths.tail())                

State          CA     FL     NC     NY     PA     TX
Date                                                
10/17/20_x   50.0  188.0  198.0   92.0  145.0  175.0
10/18/20_x   35.0  118.0  219.0   71.0  108.0  200.0
10/19/20_x  121.0   79.0  112.0   51.0   78.0   88.0
10/20/20_x  101.0  170.0  151.0   62.0  122.0  175.0
10/21/20_x   86.0  104.0    0.0  104.0  111.0  188.0
State        CA   FL   NC   NY   PA   TX
Date                                    
10/17/20_y  1.0  4.0  2.0  1.0  1.0  3.0
10/18/20_y  1.0  2.0  1.0  1.0  0.0  1.0
10/19/20_y  1.0  2.0  0.0  1.0  3.0  0.0
10/20/20_y  1.0  4.0  5.0  1.0  3.0  2.0
10/21/20_y  4.0  5.0  0.0  0.0  2.0  4.0


### Changes date format for better split across the weeks. 

In [13]:
#change date format for cases
MyStateCases  = Norm_MyStateCases.T.rename(columns=lambda x: (pd.to_datetime(x.split('_')[0], errors= 'ignore')), errors='raise').T
print(MyStateCases.tail())

#change date format for deaths
MyStateDeaths  = Norm_MyStateDeaths.T.rename(columns=lambda x: (pd.to_datetime(x.split('_')[0], errors= 'ignore')), errors='raise').T
print(MyStateDeaths.tail())

MyStateCases.to_csv("State_cases.csv")
MyStateDeaths.to_csv("State_deaths.csv")

State          CA     FL     NC     NY     PA     TX
Date                                                
2020-10-17   50.0  188.0  198.0   92.0  145.0  175.0
2020-10-18   35.0  118.0  219.0   71.0  108.0  200.0
2020-10-19  121.0   79.0  112.0   51.0   78.0   88.0
2020-10-20  101.0  170.0  151.0   62.0  122.0  175.0
2020-10-21   86.0  104.0    0.0  104.0  111.0  188.0
State        CA   FL   NC   NY   PA   TX
Date                                    
2020-10-17  1.0  4.0  2.0  1.0  1.0  3.0
2020-10-18  1.0  2.0  1.0  1.0  0.0  1.0
2020-10-19  1.0  2.0  0.0  1.0  3.0  0.0
2020-10-20  1.0  4.0  5.0  1.0  3.0  2.0
2020-10-21  4.0  5.0  0.0  0.0  2.0  4.0


## Calculate weekly statistics

In [14]:
#Calculate mean number of new cases and New Deaths per week
WeeklyState_cases = MyStateCases.resample('W').mean().round(0).astype(int).rename_axis('Week')
WeeklyState_deaths = MyStateDeaths.resample('W').mean().round(0).astype(int).rename_axis('Week')


print(WeeklyState_cases)
print(WeeklyState_deaths)

State        CA   FL   NC   NY   PA   TX
Week                                    
2020-01-26    0    0    0    0    0    0
2020-02-02    0    0    0    0    0    0
2020-02-09    0    0    0    0    0    0
2020-02-16    0    0    0    0    0    0
2020-02-23    0    0    0    0    0    0
2020-03-01    0    0    0    0    0    0
2020-03-08    0    0    0    1    0    0
2020-03-15    2    0    0    4    0    0
2020-03-22    5    6    3  112    5    2
2020-03-29   17   26   12  318   32   12
2020-04-05   31   49   19  462   91   22
2020-04-12   29   50   26  488  126   33
2020-04-19   29   43   26  397  105   27
2020-04-26   44   35   28  333   99   27
2020-05-03   41   30   43  209   90   33
2020-05-10   47   30   42  139   82   37
2020-05-17   44   33   51  108   63   43
2020-05-24   52   35   64   84   61   38
2020-05-31   66   35   61   68   47   44
2020-06-07   67   52  107   54   41   52
2020-06-14   78   77  117   38   36   65
2020-06-21   89  144  118   34   33  116
2020-06-28  133 

In [15]:
State_MeanCases = WeeklyState_cases.mean().round(0)
State_MedianCases = WeeklyState_cases.median()
ModeCases = WeeklyState_cases.mode()

State_MeanDeaths = WeeklyState_deaths.mean().round(0)
State_MedianDeaths = WeeklyState_deaths.median()
ModeDeaths = WeeklyState_deaths.mode()


State_ModeDeaths = ModeDeaths.values.tolist()
State_ModeCases = ModeCases.values.tolist()



# Puts all values organized into a dataframe for better visability

In [16]:
ListCase = [["Texas",State_MeanCases[5],State_MedianCases[5],State_ModeCases[0][5]],
        ["California", State_MeanCases[0],State_MedianCases[0],State_ModeCases[0][0]],
        ["Florida", State_MeanCases[1],State_MedianCases[1],State_ModeCases[0][1]],
        ["New York", State_MeanCases[3],State_MedianCases[3],State_ModeCases[1][3]],
        ["North Carolina", State_MeanCases[2],State_MedianCases[2],State_ModeCases[0][2]],
       ["Philadelphia", State_MeanCases[4],State_MedianCases[4],State_ModeCases[0][4]]]


ListDeaths = [["Texas",State_MeanDeaths[5],State_MedianDeaths[5],State_ModeDeaths[0][5]],
        ["California", State_MeanDeaths[0],State_MedianDeaths[0],State_ModeDeaths[0][0]],
        ["Florida", State_MeanDeaths[1],State_MedianDeaths[1],State_ModeDeaths[0][1]],
        ["New York", State_MeanDeaths[3],State_MedianDeaths[3],State_ModeDeaths[0][3]],
        ["North Carolina", State_MeanDeaths[2],State_MedianDeaths[2],State_ModeDeaths[0][2]],
       ["Philadelphia", State_MeanDeaths[4],State_MedianDeaths[4],State_ModeDeaths[0][4]]]

In [17]:
StateWeeklyStatisticsCase = pd.DataFrame(ListCase, columns = ['State','Mean', 'Median', 'Mode'])

StateWeeklyStatisticsCase

State   Mean  Median  Mode
0           Texas  107.0    58.5   0.0
1      California   82.0    70.5   0.0
2         Florida  128.0    64.5   0.0
3        New York   91.0    38.0  34.0
4  North Carolina   86.0    97.5   0.0
5    Philadelphia   53.0    59.5   0.0

In [18]:
StateWeeklyStatisticsDeaths = pd.DataFrame(ListDeaths, columns = ['State','Mean', 'Median', 'Mode'])

StateWeeklyStatisticsDeaths

State  Mean  Median  Mode
0           Texas   2.0     1.0     1
1      California   2.0     2.0     2
2         Florida   3.0     2.0     2
3        New York   6.0     1.0     0
4  North Carolina   1.0     2.0     2
5    Philadelphia   2.0     1.0     1

# Analysis 
### Compared to other states, Texas has a higher weekly average of new cases right behind Florida. 
### However, Texas doesn not have a really high weekly average death compared to states like California, New York and Florida. 

# Identify Counties of high Cases and Death Rate of Texas

In [19]:
Texas = df.loc[df['State']=='TX']
Texas = Texas.set_index('County Name')

#look for recent day to see total amount of cases and order the values to see counties with high case and death rates. 
Texas

Tex_cases = Texas.filter(regex = "x")
Tex_deaths = Texas.filter(regex = "y").drop(["County Name_y"], axis=1)





In [20]:
#display last row which is the total cases and death 
LastRowCase = Tex_cases.iloc[:,-1:]
LastRowDeath = Tex_deaths.iloc[:,-1:]

#Have last row be renamed to total cases and total deaths
TotalCases = LastRowCase.rename(columns={list(LastRowCase)[0]: 'Total Cases'})
TotalDeaths = LastRowDeath.rename(columns={list(LastRowDeath)[0]: 'Total Deaths'})

#print top 7 counties of cases and deaths in Texas
print(TotalCases.sort_values(by = ['Total Cases'], ascending = False).head(7))
print()
print(TotalDeaths.sort_values(by = ['Total Deaths'], ascending = False).head(7))

                Total Cases
County Name                
Harris County        155192
Dallas County         90790
Bexar County          63807
Tarrant County        54727
El Paso County        34760
Hidalgo County        34343
Travis County         31159

                Total Deaths
County Name                 
Harris County           2750
Hidalgo County          1655
Bexar County            1385
Dallas County           1205
Cameron County           968
Tarrant County           796
El Paso County           579


### Counties with High cases: 
#### Harris county
#### Dallas county
#### Bexar county
#### Tarrant county
#### Hidalgo county

### Counties with High deaths: 
#### Harris County      =    2726
#### Hidalgo County   =    1640
#### Bexar County      =    1379
#### Dallas County     =    1190
#### Cameron County  =  962 


In [21]:
#Find population of the top 5 infected counties. 

array = ['Harris County','Dallas County','Bexar County','Tarrant County', 'Hidalgo County']

CountyPop = Texas.loc[Texas.index.isin(array)]

#get Population value of counties to use later on when normalizing. 
Harris_Pop = CountyPop.iloc[2][1]
Dallas_Pop = CountyPop.iloc[1][1]
Bexar_Pop = CountyPop.iloc[0][1]
Tarrant_Pop = CountyPop.iloc[4][1]
Hidalgo_Pop = CountyPop.iloc[3][1]

In [22]:
Counties_Cases = Tex_cases.loc[Tex_cases.index.isin(array)].T.rename_axis('Date')
Counties_Deaths = Tex_deaths.loc[Tex_deaths.index.isin(array)].T.rename_axis('Date')

In [23]:
#change date format for cases
Tex_Counties_Cases  = Counties_Cases.T.rename(columns=lambda x: (pd.to_datetime(x.split('_')[0], errors= 'ignore')), errors='raise').T
print(Tex_Counties_Cases.tail())

#change date format for deaths
Tex_Counties_Deaths  = Counties_Deaths.T.rename(columns=lambda x: (pd.to_datetime(x.split('_')[0], errors= 'ignore')), errors='raise').T
print(Tex_Counties_Deaths .tail())

County Name  Bexar County  Dallas County  Harris County  Hidalgo County  \
Date                                                                      
2020-10-17          60629          88834         153571           34179   
2020-10-18          63328          89433         153955           34179   
2020-10-19          63426          89987         154367           34179   
2020-10-20          63594          90318         154765           34179   
2020-10-21          63807          90790         155192           34343   

County Name  Tarrant County  
Date                         
2020-10-17            53128  
2020-10-18            53629  
2020-10-19            54025  
2020-10-20            54332  
2020-10-21            54727  
County Name  Bexar County  Dallas County  Harris County  Hidalgo County  \
Date                                                                      
2020-10-17           1379           1190           2726            1640   
2020-10-18           1381           119

In [24]:
#check to see if dataset is correct
Tex_Counties_Cases.tail()

County Name  Bexar County  Dallas County  Harris County  Hidalgo County  \
Date                                                                      
2020-10-17          60629          88834         153571           34179   
2020-10-18          63328          89433         153955           34179   
2020-10-19          63426          89987         154367           34179   
2020-10-20          63594          90318         154765           34179   
2020-10-21          63807          90790         155192           34343   

County Name  Tarrant County  
Date                         
2020-10-17            53128  
2020-10-18            53629  
2020-10-19            54025  
2020-10-20            54332  
2020-10-21            54727

In [25]:
#check to see if dataset is correct
Tex_Counties_Deaths.tail()

County Name  Bexar County  Dallas County  Harris County  Hidalgo County  \
Date                                                                      
2020-10-17           1379           1190           2726            1640   
2020-10-18           1381           1194           2728            1644   
2020-10-19           1381           1194           2729            1646   
2020-10-20           1383           1201           2735            1650   
2020-10-21           1385           1205           2750            1655   

County Name  Tarrant County  
Date                         
2020-10-17              782  
2020-10-18              783  
2020-10-19              784  
2020-10-20              786  
2020-10-21              796

In [26]:
#Calculates differences of cases and deaths of each day (New Cases & New Deaths)
New_Tex_Counties_Cases = Tex_Counties_Cases.diff()
New_Tex_Counties_Deaths = Tex_Counties_Deaths.diff()

print(New_Tex_Counties_Cases.tail())
print(New_Tex_Counties_Deaths.tail())

County Name  Bexar County  Dallas County  Harris County  Hidalgo County  \
Date                                                                      
2020-10-17          187.0          462.0          594.0           145.0   
2020-10-18         2699.0          599.0          384.0             0.0   
2020-10-19           98.0          554.0          412.0             0.0   
2020-10-20          168.0          331.0          398.0             0.0   
2020-10-21          213.0          472.0          427.0           164.0   

County Name  Tarrant County  
Date                         
2020-10-17            494.0  
2020-10-18            501.0  
2020-10-19            396.0  
2020-10-20            307.0  
2020-10-21            395.0  
County Name  Bexar County  Dallas County  Harris County  Hidalgo County  \
Date                                                                      
2020-10-17            3.0            3.0            8.0             3.0   
2020-10-18            2.0            4.

In [27]:
#calculate and Normalize dataset to cases per 200,000 people. 
norm_val = 200_000

#Normalize cases
Norm_MyTxCases = New_Tex_Counties_Cases.assign(Harris_County = lambda x: (x['Harris County'] / Harris_Pop) * norm_val).round(0)
Norm_MyTxCases = Norm_MyTxCases.assign(Dallas_County = lambda x: (x['Dallas County'] / Dallas_Pop) * norm_val).round(0)
Norm_MyTxCases = Norm_MyTxCases.assign(Bexar_County = lambda x: (x['Bexar County'] / Bexar_Pop) * norm_val).round(0)
Norm_MyTxCases = Norm_MyTxCases.assign(Tarrant_County = lambda x: (x['Tarrant County'] / Tarrant_Pop) * norm_val).round(0)
Norm_MyTxCases = Norm_MyTxCases.assign(Hidalgo_County = lambda x: (x['Hidalgo County'] / Hidalgo_Pop) * norm_val).round(0)


#Normalize deaths 
Norm_MyTxDeaths = New_Tex_Counties_Deaths.assign(Harris_County = lambda x: (x['Harris County'] / Harris_Pop) * norm_val).round(0)
Norm_MyTxDeaths = Norm_MyTxDeaths.assign(Dallas_County = lambda x: (x['Dallas County'] / Dallas_Pop) * norm_val).round(0)
Norm_MyTxDeaths = Norm_MyTxDeaths.assign(Bexar_County = lambda x: (x['Bexar County'] / Bexar_Pop) * norm_val).round(0)
Norm_MyTxDeaths = Norm_MyTxDeaths.assign(Tarrant_County = lambda x: (x['Tarrant County'] / Tarrant_Pop) * norm_val).round(0)
Norm_MyTxDeaths = Norm_MyTxDeaths.assign(Hidalgo_County = lambda x: (x['Hidalgo County'] / Hidalgo_Pop) * norm_val).round(0)

MyTxCases = Norm_MyTxCases.drop(['Harris County','Dallas County','Bexar County','Tarrant County', 'Hidalgo County'], axis=1)
MyTxDeaths = Norm_MyTxDeaths.drop(['Harris County','Dallas County','Bexar County','Tarrant County', 'Hidalgo County'], axis=1)


print(MyTxCases.tail())
print(MyTxDeaths.tail())   

County Name  Harris_County  Dallas_County  Bexar_County  Tarrant_County  \
Date                                                                      
2020-10-17            25.0           35.0          19.0            47.0   
2020-10-18            16.0           45.0         269.0            48.0   
2020-10-19            17.0           42.0          10.0            38.0   
2020-10-20            17.0           25.0          17.0            29.0   
2020-10-21            18.0           36.0          21.0            38.0   

County Name  Hidalgo_County  
Date                         
2020-10-17             33.0  
2020-10-18              0.0  
2020-10-19              0.0  
2020-10-20              0.0  
2020-10-21             38.0  
County Name  Harris_County  Dallas_County  Bexar_County  Tarrant_County  \
Date                                                                      
2020-10-17             0.0            0.0           0.0             0.0   
2020-10-18             0.0            0

# Daily Trend of New Cases in Texas per Million people. 

In [37]:
#plot daily Texas new Cases
trace0 = go.Scatter(x=MyStateCases.index, y=MyStateCases["TX"], 
                    name="Texas", 
                    mode="lines+markers") 



mydata = go.Data([trace0])

mylayout = go.Layout(
    title="Daily New Cases of Texas State Per Million people"
)

fig1 = go.Figure(data=mydata, layout=mylayout)

plotly.offline.iplot(fig1, filename = '')

fig1.write_image("images/DailyTrend_NewCases_Texas.png")

# Daily Trend of New Deaths of Top 5 Infected Counties in Texas per 200,000 people. 

In [38]:
#plot daily texas counties new Cases
trace0 = go.Scatter(x=MyTxCases.index, y=MyTxCases["Harris_County"], 
                    name="Harris County", 
                    mode="lines+markers") 

trace1 = go.Scatter(x=MyTxCases.index, y=MyTxCases["Dallas_County"], 
                    name="Dallas County", 
                    mode="lines+markers") 

trace2 = go.Scatter(x=MyTxCases.index, y=MyTxCases["Bexar_County"], 
                    name="Bexar County", 
                    mode="lines+markers") 

trace3 = go.Scatter(x=MyTxCases.index, y=MyTxCases["Tarrant_County"], 
                    name="Tarrant County", 
                    mode="lines+markers") 

trace4 = go.Scatter(x=MyTxCases.index, y=MyTxCases["Hidalgo_County"], 
                    name="Hidalgo County", 
                    mode="lines+markers") 

mydata = go.Data([trace0, trace1,trace2, trace3,trace4])

mylayout = go.Layout(
    title="Daily New Cases Per 100,00 people of top infected Texas Counties"
)

fig2 = go.Figure(data=mydata, layout=mylayout)

plotly.offline.iplot(fig2, filename = '')

fig2.write_image("images/DailyTrend_NewCases_TexasCounties.png")

# Daily Trend of New Deaths in Texas Per Million people. 

In [39]:
#plot daily Texas new Deaths
trace0 = go.Scatter(x=MyStateDeaths.index, y=MyStateDeaths["TX"], 
                    name="Texas", 
                    mode="lines+markers") 


mydata = go.Data([trace0])

mylayout = go.Layout(
    title="Daily New Deaths of Texas State Per Million people"
)

fig3 = go.Figure(data=mydata, layout=mylayout)

plotly.offline.iplot(fig3, filename = '')

fig3.write_image("images/DailyTrend_NewDeaths_Texas.png")

# Daily Trend of New Deaths of Top 5 Infected Counties in Texas per 200,000 people. 

In [40]:
#plot daily texas counties new death 

trace0 = go.Scatter(x=MyTxDeaths.index, y=MyTxDeaths["Harris_County"], 
                    name="Harris County", 
                    mode="lines+markers") 

trace1 = go.Scatter(x=MyTxDeaths.index, y=MyTxDeaths["Dallas_County"], 
                    name="Dallas County", 
                    mode="lines+markers") 

trace2 = go.Scatter(x=MyTxDeaths.index, y=MyTxDeaths["Bexar_County"], 
                    name="Bexar County", 
                    mode="lines+markers") 

trace3 = go.Scatter(x=MyTxDeaths.index, y=MyTxDeaths["Tarrant_County"], 
                    name="Tarrant County", 
                    mode="lines+markers") 

trace4 = go.Scatter(x=MyTxDeaths.index, y=MyTxDeaths["Hidalgo_County"], 
                    name="Hidalgo County", 
                    mode="lines+markers") 

mydata = go.Data([trace0, trace1,trace2, trace3,trace4])

mylayout = go.Layout(
    title="Daily New Deaths Per 100,00 people of top infected Texas Counties"
)

fig4 = go.Figure(data=mydata, layout=mylayout)

plotly.offline.iplot(fig4, filename = '')

fig4.write_image("images/DailyTrend_NewDeaths_TexasCounties.png")